### ISSUE TITLE
also accept .yml (not just .yaml) as configuration extension

### ISSUE THREAD

**Issue by matleh** [pyinvoke/invoke#232]
Most tools I know use ".yml" as extension for YAML files (docker-compose for example).
Invoke currently wants ".yaml". 
If Invoke would also accept ".yml" this would spare users who are used to yml an extra thought and file-rename round-trip.

No big deal but nice to have.

**Comment by bitprophet** [comment]
I feel like I had a specific reason for not honoring both, but cannot remember offhand what that was (or if I am misremembering). May have been due to an earlier use of a third party lib in the config stuff, for example (which no longer applies).

OK...I looked, the ticket for that is #147, the specific bits where @sophacles and I bantered about it (among other things - just search for "yml" within these comments), are:

* https://github.com/pyinvoke/invoke/issues/147#issuecomment-55349915
* https://github.com/pyinvoke/invoke/issues/147#issuecomment-55432423
* https://github.com/pyinvoke/invoke/issues/147#issuecomment-55476683
* https://github.com/pyinvoke/invoke/issues/147#issuecomment-55499814

Seems like the primary reason besides "it's cleaner to only support one" is that we'd have to declare a separate "load order" for yml vs yaml, should the user have both in the same location for whatever reason. (Alternately, we could just explode angrily as per [the Zen of Python](http://zen-of-python.info/in-the-face-of-ambiguity-refuse-the-temptation-to-guess.html#12).)

I haven't looked in the config junk basically since that ticket completed...if it's reasonably easy to actually implement this (preferably the explosion option), I would probably be +1 on it, as then the benefits would outweigh the negatives.

**Comment by gtback** [comment]
It seems to be as easy as:

```diff
diff --git a/invoke/config.py b/invoke/config.py
index 9ab852e..1ca28fc 100644
--- a/invoke/config.py
+++ b/invoke/config.py
@@ -278,7 +278,7 @@ class Config(DataProxy):
             prefix.
         """
         # Config file suffixes to search, in preference order.
-        self._file_suffixes = ('yaml', 'json', 'py')
+        self._file_suffixes = ('yaml', 'yml', 'json', 'py')
 
         # Technically an implementation detail - do not expose in public API.
         # Stores merged configs and is accessed via DataProxy.
@@ -557,6 +557,8 @@ class Config(DataProxy):
         with open(path) as fd:
             return yaml.load(fd)
 
+    _load_yml = _load_yaml
+
     def _load_json(self, path):
         with open(path) as fd:
             return json.load(fd)
```
(plus documentation changes, of course). I'm happy to make the changes  if @bitprophet concurs.

**Comment by bitprophet** [comment]
That seems pretty likely, @gtback - I'd probably merge that PR (if it has doc/changelog changes as you mentioned :)) - thanks!

**Comment by bitprophet** [comment]
Linked commit was a bit messy, was part impl + changelog. Previous commit included all the tests not discussed herein: 563cecbad0fa2efcdb6b823103fe22901837ca1a
